app/experimenter/docs/adrs/0002-use-react-bootstrap-cra-gql-for-nimbus.md

```diff
@@ -0,0 +1,113 @@
+
+* Status: proposed
+* Deciders: Lauren Zugai, Jody Heavener, [TBD]
+* Date: 2020-09-08
+
+## Context and Problem Statement
+
+The front-end for Experimenter is currently composed of two applications. A more mature application for Normandy experiments, found at `app/experimenter/static/core`, is composed of a mix of Django templates, vanilla JS, jQuery, and React components. A newer React application for Nimbus experiments (previously named Rapid experiments) can be found at `app/experimenter/static/rapid`.
+
+With the decision made to build the Nimbus console as part of the Experimenter codebase ([see ADR #0001](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md)), it was determined we likely want a "new front-end." This ADR, while slightly atypical as it does not serve to document a single architectural change, outlines a proposed directory tree change regarding the front-end, a proposed new front-end stack, how it compares to the current `rapid` stack, and how we will benefit from these changes.
+
+## Decision Drivers
+
+Drivers for directory tree changes:
+* A more obvious separation of concerns for:
+  * the front-end and back-end
+  * legacy front-end code and Nimbus front-end code
+
+Drivers for the proposed new stack:
+* Better development tooling
+* A more cohesive fit into the FxA team's stack and processes
+* Development speed
+
+Drivers for starting a new application:
+* Time needed and ease of refactoring the current `rapid` React application vs time needed to start with a new application
+
+## Proposed Changes
+
+### App Directory Tree
+
+Currently, a `static/` directory contains the front-end applications with some shared assets, and a `templates/` directory lives adjacent to it.  To provide a more obvious separation of concerns in the directory tree:
+
+* Rename `static/` to `legacy-ui/` since `static/rapid/` will become obsolete and `static/core/` will be the legacy Normandy platform
+* Move `templates/` into `legacy-ui/`
+* Use `create-react-app` to create a new React application in `app/experimenter/nimbus-ui/`
+
+
+### Front-End Stack
+
+* Continue using React, but:
+  * Use the `create-react-app` toolchain
+  * Use Typescript over Proptypes for more strict typing
+  * Use GraphQL for API calls and `apollo-client-cache` for global state management
+  * Use `reach-router` instead of `react-router`
+  * Use directories for components instead of filenames
+    * The Storybook and test file will live adjacent to the component `index.tsx` file. This allows all associated component files to live in the same directory and makes imports shorter.
+* Continue using Bootstrap (`react-bootstrap`), but:
+  * Add stylelint for SCSS linting overrides
+  * Use breakpoints established in FxA
+* Utilize Storybook for browsing the UI in various states without running the stack
+* Add Sentry to catch client-side errors
+
```

Comment:
I see the existing `rapid` setup is using `experiments`, `forms` and `pages` to organize components by type. This type of organization was recently discussed for an FxA React project, so I'm curious if we'll be employing the same structure with the new app. My personal preference is a flat directory of well-named components.

Reply:
While I'm more indifferent to this, it was clear the FxA team likes the flat structure, so I can include that as an additional point under "Use directories for components instead of filenames" above

Reply:
I'm curious if @tiftran or @jaredlockhart have a strong preference here. Check out our [component directory structure](https://github.com/mozilla/fxa/tree/main/packages/fxa-settings/src/components) for `fxa-settings`, we're generally using a `nounDescriptor` approach to keep similar files like Inputs and Pages grouped together in the tree but keep all of them one directory deep under `components`. Makes it easy to scan what's available while keeping imports simple.

FWIW we've separated "Pages" out in our Storybook which [you can see here](https://mozilla-fxa.github.io/storybooks/commits/5f2ad51fadd80ea3afef726ea7d66b2eecdc8987/fxa-settings/index.html?path=/story/pages-settings--cold-start).

Reply:
Sweet, yeah I think `directory/index.tsx` especially makes sense with the introduction of Storybook files. Keeping the test file and Storybook file adjacent to the component file has been super handy, and we've also been adding images to these directories when they're only used in that component (and then we move them into a [shared `images/` directory](https://github.com/mozilla/fxa/tree/main/packages/fxa-react/images) when needed). The [Settings Nav](https://github.com/mozilla/fxa/tree/main/packages/fxa-settings/src/components/Nav) is a good example.